File: .Net/Home Assistant/HomeAssistant.Common/Models/SmartDevice.cs
using HomeAssistant.Common.Models;
using System.ComponentModel.DataAnnotations;

namespace HomeAssistant.Common.Models
{
    public class SmartDevice
    {
        [Key]
        public Guid DeviceId { get; set; }
        public string DevicePassword { get; set; }
        public string Name { get; set; }
        public int RoomIndex { get; set; } 
        public bool IsOn { get; set; }
    }

    public class RoomIndex
    {
        [Key]
        public Guid Id { get; set; }
        public int Index { get; set; }

        public Guid DeviceId { get; set; }
        public SmartDevice SmartDevice { get; set; }
    }
}

File: Flutter/Home Assistant/lib/Features/Home/Models/SmartDevice.dart
class SmartDevice {
  final String deviceId;
  final String name;
  final String image;
  final int roomIndex;
  bool isOn;
  SmartDevice(this.deviceId, this.name, this.image, this.roomIndex, this.isOn);

  Map<String, dynamic> toJson() {
    return {
      'deviceId': deviceId,
      'isOn': isOn,
    };
  }
}
